### PR TITLE
Fix: generate correct URIs for entities with "/"

### PR DESCRIPTION
**Pull request by meretp**
Some entitities contain slash notation to reference elements in different namespaces which lead to incorrect URIs. With this PR besides `:` also `/` are taken into account and treated different when generating URIs.

### CODE REVIEW

File: spec_parser/utils.py [hunk 1]
```diff
@@ -268,26 +268,28 @@ def _extract_entries(self, entry_list):
...
-
-        if len(splitted) > 2:
+        if getattr(self, "spec", None) is None:
             return Literal(entity)
 
-        if getattr(self, "spec", None) is None:
+        if ":" in entity:
+            namespace_and_entity = re.split(r":", entity)
+        elif "/" in entity:
+            namespace_and_entity = re.split(r"/", entity)[1:]
```
Comment (maxhbr) [inline spec_parser/utils.py:277]: this drops the first empty entry? Maybe instead filter for `!= ""` or only drop if is empty.

**Review by maxhbr — APPROVED** [review]
Looks ok and is probably correct. But this would hugely benefit from tests!

File: spec_parser/utils.py (hunk 1 above)
Comment (meretp) [reply spec_parser/utils.py]: Done

**Comment by zvr** [comment]
I find this implementation too hack-ish...

What about (pseudo-code):
```
    if ":" in entity:
        log_error("don't use colons, use slashes")
   ns_and_ent = entity.lstrip('/').rsplit('/', 1)
   if len(ns_and_ent) == 1:
      ent = ns_and_ent
   else:
     [ns, ent] = ns_and_ent
```

**Comment by meretp** [comment]
@zvr I think it makes sense as the external namespaces are also part of `self.spec.rdf_dict` and it fits to the generation of the uri using `_namespace` and `_entity`. I added another commit to update the code, please have a look if this implementation is better.

**Review by zvr — APPROVED** [review]

**Pull request merged**